### PR TITLE
Rename Request protocol to RequestType for consistency with Swift standard library

### LINKED ISSUE #79: Name of Request protocol
In Swift standard library, protocol names follow naming conventions like ...able or ...Type. For example, `Equiatable`, `SequenceType` and `ErrorType`. I think `RequestType` is more suitable for the name of request protocol, and this change should be applied before 1.0.0 is released.

@ikesyo What do you think?

### PR DESCRIPTION
**Pull request by ikesyo**
Resolves #79.

### CODE REVIEW

File: APIKitTests/APITests.swift [hunk 1]
```diff
@@ -3,7 +3,7 @@ import APIKit
 import XCTest
 import OHHTTPStubs
 
-protocol MockAPIRequest: Request {
+protocol MockAPIRequest: RequestType {
```
Comment (ikesyo) [inline APIKitTests/APITests.swift]: Should we rename it to `MockAPIRequestType`?

File: Demo.playground/Contents.swift [hunk 2]
```diff
@@ -5,7 +5,7 @@ import APIKit
 XCPSetExecutionShouldContinueIndefinitely()
 
 //: Step 1: Define request protocol
-protocol GitHubRequest: Request {
+protocol GitHubRequest: RequestType {
```
Comment (ikesyo) [inline Demo.playground/Contents.swift]: Should we rename it to `GitHubRequestType`?

File: APIKitTests/APITests.swift (hunk 1 above)
Comment (ishkawa) [inline APIKitTests/APITests.swift]: Sure :smile:

File: Demo.playground/Contents.swift (hunk 2 above)
Comment (ishkawa) [inline Demo.playground/Contents.swift]: Same as above.

**Comment by ikesyo** [comment]
Done! :zap:

**Comment by ishkawa** [comment]
:+1:

**Pull request merged**